packages/shared-ux/chrome/navigation/src/ui/v2/use_init_navnode.ts

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ChromeNavLink } from '@kbn/core-chrome-browser';
+import { useCallback, useEffect, useMemo, useRef } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+
+import { useNavigation as useNavigationServices } from '../../services';
+import { InternalNavigationNode, NodeProps, RegisterFunction, UnRegisterFunction } from './types';
+import { useRegisterTreeNode } from './use_register_tree_node';
+
+function getIdFromNavigationNode({ id: _id, link, title }: NodeProps): string {
+  const id = _id ?? link;
+
+  if (!id) {
+    throw new Error(`Id or link prop missing for navigation item [${title}]`);
+  }
+
+  return id;
+}
+
+function createInternalNavNode(
+  id: string,
+  _navNode: NodeProps,
+  deepLinks: Readonly<ChromeNavLink[]>
+): InternalNavigationNode | null {
+  const { children, onRemove, link, ...navNode } = _navNode;
+  const deepLink = deepLinks.find((dl) => dl.id === link);
+  const isLinkActive = isNodeActive({ link, deepLink });
+
+  const titleFromDeepLinkOrChildren = typeof children === 'string' ? children : deepLink?.title;
+  let title = navNode.title ?? titleFromDeepLinkOrChildren;
+
+  if (!title || title.trim().length === 0) {
+    if (isLinkActive) {
+      throw new Error(`Title prop missing for navigation item [${id}]`);
+    } else {
+      // No title provided but the node is disabled, so we can safely set it to an empty string
+      title = '';
+    }
+  }
+
+  if (!isLinkActive) {
+    return null;
+  }
+
+  return {
+    ...navNode,
+    id,
+    title,
+    deepLink,
+  };
+}
+
+function isNodeActive({ link, deepLink }: { link?: string; deepLink?: ChromeNavLink }) {
+  if (link && !deepLink) {
+    // If a link is provided, but no deepLink is found, don't render anything
+    return false;
+  }
+  return true;
```

Comment:
I think the choice of `Active` in the naming could get confusing. We will more often than not refer to an "active" link as one that matches the current URL and should be styled as "active." 

Maybe `isRenderable` would be a better name?

Reply:
Good point. I renamed it to `isNodeVisible` on my current PR. If you don't mind I'll bring the change there 👍